src/kernel/bitcoinkernel.h

```diff
@@ -0,0 +1,1197 @@
+ * under the constraints specified by flags. If the
+ * `kernel_SCRIPT_FLAGS_VERIFY_WITNESS` flag is set in the flags bitfield, the
+ * amount parameter is used. If the taproot flag is set, the spent outputs
+ * parameter is used to validate taproot transactions.
+ *
+ * @param[in] script_pubkey     Non-null, script pubkey to be spent.
+ * @param[in] amount            Amount of the script pubkey's associated output. May be zero if
+ *                              the witness flag is not set.
+ * @param[in] tx_to             Non-null, transaction spending the script_pubkey.
+ * @param[in] spent_outputs     Nullable if the taproot flag is not set. Points to an array of
+ *                              outputs spent by the transaction.
+ * @param[in] spent_outputs_len Length of the spent_outputs array.
+ * @param[in] input_index       Index of the input in tx_to spending the script_pubkey.
+ * @param[in] flags             Bitfield of kernel_ScriptFlags controlling validation constraints.
+ * @param[out] status           Nullable, will be set to an error code if the operation fails.
+ *                              Should be set to kernel_SCRIPT_VERIFY_OK.
+ * @return                      True if the script is valid.
+ */
+BITCOINKERNEL_API bool BITCOINKERNEL_WARN_UNUSED_RESULT kernel_verify_script(
+    const kernel_ScriptPubkey* script_pubkey,
+    int64_t amount,
+    const kernel_Transaction* tx_to,
+    const kernel_TransactionOutput** spent_outputs, size_t spent_outputs_len,
+    unsigned int input_index,
+    unsigned int flags,
+    kernel_ScriptVerifyStatus* status
+) BITCOINKERNEL_ARG_NONNULL(1, 3);
+
+///@}
+
+/** @name Logging
+ * Logging-related functions.
+ */
+///@{
+
+/**
+ * @brief This disables the global internal logger. No log messages will be
+ * buffered internally anymore once this is called and the buffer is cleared.
+ * This function should only be called once and is not thread or re-entry safe.
+ * Log messages will be buffered until this function is called, or a logging
+ * connection is created.
+ */
+BITCOINKERNEL_API void kernel_disable_logging();
+
+/**
+ * @brief Set the log level of the global internal logger. This does not
+ * enable the selected categories. Use `kernel_enable_log_category` to start
+ * logging from a specific, or all categories. This function is not thread
+ * safe. Mutiple calls from different threads are allowed but must be
+ * synchronized. This changes a global setting and will override settings for
+ * all existing `kernelLoggingConnection instances.
+ *
+ * @param[in] category If kernel_LOG_ALL is chosen, all messages at the specified level
+ *                     will be logged. Otherwise only messages from the specified category
+ *                     will be logged at the specified level and above.
+ * @param[in] level    Log level at which the log category is set.
+ */
+BITCOINKERNEL_API void kernel_add_log_level_category(const kernel_LogCategory category, kernel_LogLevel level);
```

Comment:
What is the rationale behind requiring to first add, and then {enable,disable} the category? An alternative would be a single `kernel_set_log_level_category`, which immediately "enables" (I think it's a strange term anyway) the category at the given level. "Disabling" would be achieved by calling `kernel_set_log_level_category` again with a higher (i.e. less granular) level, again taking effect immediately.

I can't think of any use cases that require separating this in 3 functions? I think it would simultaneously be more intuitive and ergonomic, and probably also less code?

Reply:
This should just mirror the internal code at the moment, but I agree that it is not really useful to split this up. Will see if I can consolidate this.